### DISCUSSION TITLE
Wanna make datetime type column precision at second

### DISCUSSION THREAD

**Discussion by sherrrrr** [sqlalchemy/sqlalchemy#10589] [Usage Questions]
Is there a way to make datetime column read from python object at second precision?
For example, when set a value at datetime field, make it like:`value = value.replace(microsecond=0)`.

Reason:
```
@transaction()
def test():
    user = User(create_time=datetime.datetime.now())
    db.session.add(user)
    db.session.flush()
    print(user.created_at)  # datetime.datetime(2023, 11, 6, 23, 59, 59, 903532)
    return user

user = test()  # updating its create_time like refresh from mysql
print(user.created_at)  # datetime.datetime(2023, 11, 7, 0, 0, 0)   # become next day, bad
```

and I don't want to add `db.session.refresh(user)` after every `db.session.flush()`. so I want to make this at column class level, and using setter function for every datetime column at every table class is not acceptable.

**Comment by CaselIT** [comment]
Hi,

You can create a type decorator and use that type instead of the datetime. https://docs.sqlalchemy.org/en/20/core/custom_types.html#augmenting-existing-types

if you use the v2 annotated declarative you can configure the type mat to automatically use the decorator https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#orm-declarative-mapped-column-type-map
